### PR TITLE
Updates Open in Tor view in location bar.

### PR DESCRIPTION
**Pull request by mkarolin**
Resolves https://github.com/brave/brave-browser/issues/39383

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by mkarolin** [comment]
@simonhong, I think putting this action on the left of all others might be a bit complicated so I was thinking of leaving that out as a follow up. If you can think of an easy implementation for that, please, let me know.

**Comment by simonhong** [comment]
> @simonhong, I think putting this action on the left of all others (as @aguscruiz proposed in [brave/brave-browser#39383 (comment)](https://github.com/brave/brave-browser/issues/39383#issuecomment-2191576063)) might be a bit complicated so I was thinking of leaving that out as a follow up. If you can think of an easy implementation for that, please, let me know.

Tried to set it to left most of trailing view. WDYT?
image

File: browser/ui/views/location_bar/onion_location_view.cc [hunk 1]
```diff
@@ -43,10 +43,15 @@
 
 namespace {
 
-constexpr SkColor kOpenInTorBg = SkColorSetRGB(0x6a, 0x37, 0x85);
-constexpr SkColor kIconColor = SkColorSetRGB(0xf0, 0xf2, 0xff);
+constexpr SkColor kOpenInTorBg = SkColorSetRGB(0x8c, 0x30, 0xbb);
+constexpr SkColor kIconColor = SK_ColorWHITE;
 constexpr SkColor kTextColor = SK_ColorWHITE;
-constexpr int kIconSize = 12;
+constexpr int kFontSize = 18;
```
[the remark this answers is not in the document]
Comment (mkarolin) [reply browser/ui/views/location_bar/onion_location_view.cc]: Fixed in [eabef10](https://github.com/brave/brave-core/pull/24447/commits/eabef109439ba8ec36c530c85366a3fef599a27f)

**Comment by mkarolin** [comment]
> Tried to set it to left most of trailing view. WDYT?

Thank @simonhong, I think that looks great. 😻

I added one little simplification in [cd60d25](https://github.com/brave/brave-core/pull/24447/commits/cd60d25f3e93c144eea19e7993681d10007f89a7)

**Review by simonhong — APPROVED** [review]
++

**Review by goodov — APPROVED** [review]

**Review by goodov — APPROVED** [review]

**Review by bsclifton — APPROVED** [review]
string reviewers ++ 😄👍

**Comment by GeetaSarvadnya** [comment]
Verification **PASSED** on

```
Brave | 1.69.90 Chromium: 127.0.6533.26 (Official Build) nightly (64-bit)
-- | --
Revision | f28a94aea02a03674fe24a352cb52453b6132be2
OS | Windows 10 Version 22H2 (Build 19045.4529)
```
Verified the comments from https://github.com/brave/brave-core/pull/24447#issuecomment-2204904575
Confirmed that `Open in Tor` button is updated as per the design on the bave.com and mail.protonmail.com 

Example | Example
----------|----------
image | image
image | image

**Comment by GeetaSarvadnya** [comment]
@mkarolin @simonhong: @rebron wants to retain logo only not text as per the slack communication https://bravesoftware.slack.com/archives/CHGKGMHDJ/p1720472348031839?thread_ts=1720455221.114959&cid=CHGKGMHDJ

**Comment by simonhong** [comment]
> @mkarolin @simonhong: @rebron wants to retain logo only not text as per the slack communication https://bravesoftware.slack.com/archives/CHGKGMHDJ/p1720472348031839?thread_ts=1720455221.114959&cid=CHGKGMHDJ

@mkarolin pushed PR for it - https://github.com/brave/brave-core/pull/24561

**Pull request merged**